include/irr/builtin/material_compiler/glsl/common.glsl

```diff
@@ -0,0 +1,1386 @@
+		i.TdotV,
+		i.BdotV,
+		ax, ax2, ay, ay2
+	);
+}
+
+//beckmann
+float irr_glsl_beckmann_height_correlated_cos_eval_DG(in float NdotH, in float NdotL, in float NdotV, in float a2)
+{
+	float NdotH2 = NdotH*NdotH;
+	float NdotL2 = NdotL*NdotL;
+	float NdotV2 = NdotV*NdotV;
+
+	return irr_glsl_beckmann_height_correlated_cos_eval_DG_wo_clamps(NdotH2, NdotL2, NdotV2, a2);
+}
+float irr_glsl_beckmann_height_correlated_cos_eval_DG(in irr_glsl_LightSample s, in irr_glsl_IsotropicMicrofacetCache uf, in irr_glsl_IsotropicViewSurfaceInteraction i, in float a2)
+{
+	return irr_glsl_beckmann_height_correlated_cos_eval_DG(uf.NdotH, s.NdotL, i.NdotV, a2);
+}
+
+float irr_glsl_beckmann_aniso_height_correlated_cos_eval_DG(in float NdotH, in float TdotH, in float BdotH, in float NdotL, in float TdotL, in float BdotL, in float NdotV, in float TdotV, in float BdotV, in float ax, in float ax2, in float ay, in float ay2)
+{
+	return irr_glsl_beckmann_aniso_height_correlated_cos_eval_DG_wo_clamps(
+		NdotH*NdotH,
+		TdotH*TdotH,
+		BdotH*BdotH,
+		NdotL*NdotL,
+		TdotL*TdotL,
+		BdotL*BdotL,
+		NdotV*NdotV,
+		TdotV*TdotV,
+		BdotV*BdotV,
+		ax, ax2, ay, ay2
+	);
+}
+float irr_glsl_beckmann_aniso_height_correlated_cos_eval_DG(in irr_glsl_LightSample s, in irr_glsl_AnisotropicMicrofacetCache uf, in irr_glsl_AnisotropicViewSurfaceInteraction i, in float ax, in float ax2, in float ay, in float ay2)
+{
+	return irr_glsl_beckmann_aniso_height_correlated_cos_eval_DG(
+		uf.isotropic.NdotH,
+		uf.TdotH,
+		uf.BdotH,
+		s.NdotL,
+		s.TdotL,
+		s.BdotL,
+		i.isotropic.NdotV,
+		i.TdotV,
+		i.BdotV,
+		ax, ax2, ay, ay2
+	);
+}
+
+//blinn-phong
+float irr_glsl_blinn_phong_cos_eval_DG(in float NdotH, in float NdotV, in float NdotL, in float n, in float a2)
+{
+	return irr_glsl_blinn_phong_cos_eval_DG_wo_clamps(NdotH, NdotV*NdotV, NdotL*NdotL, n, a2);
+}
+float irr_glsl_blinn_phong_cos_eval_DG(in irr_glsl_LightSample s, in irr_glsl_IsotropicMicrofacetCache uf, in irr_glsl_IsotropicViewSurfaceInteraction i, in float n, in float a2)
+{
+	return irr_glsl_blinn_phong_cos_eval_DG(uf.NdotH, i.NdotV, s.NdotL, n, a2);
+}
+
+float irr_glsl_blinn_phong_cos_eval_DG(in float NdotH, in float TdotH, in float BdotH, in float NdotL, in float TdotL, in float BdotL, in float NdotV, in float TdotV, in float BdotV, in float nx, in float ny, in float ax2, in float ay2)
+{
+	return irr_glsl_blinn_phong_cos_eval_DG_wo_clamps(NdotH, NdotH * NdotH, TdotH * TdotH, BdotH * BdotH, TdotL * TdotL, BdotL * BdotL, TdotV * TdotV, BdotV * BdotV, NdotV * NdotV, NdotL * NdotL, nx, ny, ax2, ay2);
+}
+float irr_glsl_blinn_phong_cos_eval_DG(in irr_glsl_LightSample s, in irr_glsl_AnisotropicMicrofacetCache uf, in irr_glsl_AnisotropicViewSurfaceInteraction i, in float nx, in float ny, in float ax2, in float ay2)
+{
+	return irr_glsl_blinn_phong_cos_eval_DG(
+		uf.isotropic.NdotH,
+		uf.TdotH,
+		uf.BdotH,
+		s.NdotL,
+		s.TdotL,
+		s.BdotL,
+		i.isotropic.NdotV,
+		i.TdotV,
+		i.BdotV,
+		nx, ny, ax2, ay2
+	);
+}
```

Comment:
what's all this doing here?

Reply:
theres no builtins with struct parameters in our glsl lib

Reply:
you wont need them https://github.com/buildaworldnet/IrrlichtBAW/pull/554/files#r526504743